pkg/auth/globalrole.go

```diff
@@ -65,9 +65,39 @@ func (g *GlobalRoleResolver) ClusterRulesFromRole(gr *v3.GlobalRole) ([]rbacv1.P
 		}
 		rules = append(rules, templateRules...)
 	}
+
 	return rules, nil
 }
 
+func (g *GlobalRoleResolver) FleetWorkspacePermissionsResourceRulesFromRole(gr *v3.GlobalRole) []rbacv1.PolicyRule {
+	if gr == nil {
+		return nil
+	}
+
+	var rules []rbacv1.PolicyRule
+	if gr.InheritedFleetWorkspacePermissions.ResourceRules != nil {
+		rules = append(rules, gr.InheritedFleetWorkspacePermissions.ResourceRules...)
+	}
+
+	return rules
+}
+
+func (g *GlobalRoleResolver) FleetWorkspacePermissionsWorkspaceVerbsFromRole(gr *v3.GlobalRole) []rbacv1.PolicyRule {
```

Comment:
I don't understand what you mean here. Could you elaborate on this, please?

Reply:
Sure - [Here](https://github.com/rancher/webhook/blob/4e47fcb77bd14e000b35fa486e9a1fd71f60994b/pkg/auth/globalrole.go#L51) is the relevant logic. Basically, even though RestrictedAdmin has no permissions in these fields, we need to give the appropriate permissions so that RA can give this role to others. But the RA doesn't have/doesn't use this field, so you need special logic here to handle that.

Reply:
done. see https://github.com/rancher/webhook/pull/348#discussion_r1639880143